src/plugins/action_trim_responses/src/databaseMethods.ts

```diff
@@ -0,0 +1,77 @@
+const databaseMethods = (DBConnect: any) => ({
+  getAllApplicationResponses: async (applicationId: number) => {
```

Comment:
I was wondering, what if some other plugin or the main code needs to use this query?

Reply:
Yeah, that was the intention when we started, hence we created a generic Database connection object with all the queries in it, thinking that plugins would be re-using a lot of the same database queries. However, as it's turned out, pretty much every Action plugin has required 3-4 *new*, unique queries not used by any other and so we've got a PostgresConnect class with loads of queries that are mostly only used by one plugin. Which is what prompted me to suggest that we should couple them with their respective plugins.

There are a few of the queries that are common, or used by the core server module (triggers/actions mostly), so these should be left as is. 

From just having a quick look, I'd say everything above `createUser` (in `postGresConnect.ts`) should be left where it is, and anything below that that is mostly plugin-specific (except where there's a specific comment to the contrary).

It's probably best to work backwards, from the bottom of postGresConnect and work up. For each database method, a global search should tell you how many places its used. But yeah, probably just do one or two and we'll make sure the others all agree on the structure before going too far.

The good thing about this is that we'll be able to easily know that any query in postGresConnect shouldn't be modified, whereas plug-in specific ones can be as they'll only affect that plugin. We've almost run into a problem a couple of times in modifying a database query for its functionality in the core, only to discover that another plugin is using it and that would mess it up. So yeah, it'll be really handy to have this better organised.

Reply:
Maybe later we can have these methods in individual plug-ins and still use them through DBConnect with interfaces, but that would need the plug-ins to be available to require js. Could be a topic for discussion and refactor later on.